src/autoCollection/metrics/nativePerformance.ts

```diff
@@ -0,0 +1,192 @@
+import { Meter, ObservableGauge, ObservableResult, Histogram } from "@opentelemetry/api-metrics";
+import { GarbageCollectionType, NativeMetricsCounter } from "../../declarations/constants";
+import { Logger } from "../../library/logging";
+import { IBaseConfig, IDisabledExtendedMetrics } from "../../library/configuration/interfaces";
+
+
+export class AutoCollectNativePerformance {
+    private _emitter: any;
+    private _metricsAvailable: boolean; // is the native metrics lib installed
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+    private _handle: NodeJS.Timer;
+    private _meter: Meter;
+    private _collectionInterval: number = 15000; // 15 seconds
+    private _disabledMetrics: IDisabledExtendedMetrics = {};
+    private _eventLoopHistogram: Histogram;
+    private _garbageCollectionScavenge: Histogram;
+    private _garbageCollectionMarkSweepCompact: Histogram;
+    private _garbageCollectionIncrementalMarking: Histogram;
+    private _heapMemoryTotalGauge: ObservableGauge;
+    private _heapMemoryUsageGauge: ObservableGauge;
+    private _memoryUsageNonHeapGauge: ObservableGauge;
+
+
+    constructor(meter: Meter) {
+        this._meter = meter;
+        this._eventLoopHistogram = this._meter.createHistogram(NativeMetricsCounter.EVENT_LOOP_CPU);
+        this._garbageCollectionScavenge = this._meter.createHistogram(NativeMetricsCounter.GARBAGE_COLLECTION_SCAVENGE);
+        this._garbageCollectionMarkSweepCompact = this._meter.createHistogram(NativeMetricsCounter.GARBAGE_COLLECTION_SWEEP_COMPACT);
+        this._garbageCollectionIncrementalMarking = this._meter.createHistogram(NativeMetricsCounter.GARBAGE_COLLECTION_INCREMENTAL_MARKING);
+        this._heapMemoryTotalGauge = this._meter.createObservableGauge(NativeMetricsCounter.HEAP_MEMORY_TOTAL);
+        this._heapMemoryUsageGauge = this._meter.createObservableGauge(NativeMetricsCounter.HEAP_MEMORY_USAGE);
+        this._memoryUsageNonHeapGauge = this._meter.createObservableGauge(NativeMetricsCounter.MEMORY_USAGE_NON_HEAP);
+    }
+
+    /**
+     * Start instance of native metrics agent.
+     *
+     * @param {boolean} isEnabled
+     * @memberof AutoCollectNativePerformance
+     */
+    public enable(
+        isEnabled: boolean,
+        disabledMetrics: IDisabledExtendedMetrics = {}
+    ): void {
+        if (this._metricsAvailable == undefined && isEnabled && !this._isInitialized) {
```

Comment:
Probably, should check that it's not already enabled, just in case someone calls this twice.

A issue came up in the JS Sig Bug triage yesterday where this can occur and break stuff -- it got tagged as a P3 to fix at some point.

Reply:
Oh, I see your using the _isInitialized for that...